### ISSUE TITLE
Value out of range (expected to be in range of [-1, 0], but got 1)

### ISSUE THREAD

**Issue by VirajBagal** [PyTorchLightning/pytorch-lightning#3185]
## Value out of range (expected to be in range of [-1, 0], but got 1)
**Exception in device=TPU:5: torch_xla/csrc/helpers.cpp:97 : Check failed: min_shape_dim <= dim && dim <= max_shape_dim**

Following is the stack trace when I am using all 8 TPU cores on Kaggle. The exact same code with 

> sync_dist=False 

works completely fine on Kaggle GPU.

 

> Value out of range (expected to be in range of [-1, 0], but got 1)Exception in device=TPU:5: torch_xla/csrc/helpers.cpp:97 : Check failed: min_shape_dim <= dim && dim <= max_shape_dim 
> *** Begin stack trace ***
> 	tensorflow::CurrentStackTrace()
> 	torch_xla::XlaHelpers::GetCanonicalDimensionIndex(long long, long long)
> 	torch_xla::XlaHelpers::MakeTransposePermutation(long long, long long, long long)
> 	torch_xla::XLATensor::transpose(torch_xla::XLATensor const&, long long, long long)
> 	torch_xla::AtenXlaType::t(at::Tensor const&)
> 	c10::impl::wrap_kernel_functor_unboxed_<c10::impl::detail::WrapFunctionIntoRuntimeFunctor_<at::Tensor (*)(at::Tensor const&), at::Tensor, c10::guts::typelist::typelist<at::Tensor const&> >, at::Tensor (at::Tensor const&)>::call(c10::OperatorKernel*, at::Tensor const&)
> 	
> 	at::t(at::Tensor const&)
> 	
> 	
> 	
> 	at::Tensor::t() const
> 	
> 	_PyMethodDef_RawFastCallKeywords
> 	_PyMethodDescr_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyEval_EvalCodeWithName
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyEval_EvalCodeWithName
> 	_PyFunction_FastCallDict
> 	_PyObject_Call_Prepend
> 	PyObject_Call
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallDict
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallDict
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallDict
> 	_PyObject_Call_Prepend
> 	
> 	_PyObject_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyEval_EvalCodeWithName
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyEval_EvalCodeWithName
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyEval_EvalCodeWithName
> 	_PyFunction_FastCallDict
> 	_PyEval_EvalFrameDefault
> 	_PyEval_EvalCodeWithName
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyEval_EvalCodeWithName
> 	PyEval_EvalCodeEx
> 	PyEval_EvalCode
> 	
> 	_PyMethodDef_RawFastCallKeywords
> 	_PyCFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyGen_Send
> 	_PyEval_EvalFrameDefault
> 	_PyGen_Send
> 	_PyEval_EvalFrameDefault
> 	_PyGen_Send
> 	_PyMethodDef_RawFastCallKeywords
> 	_PyMethodDescr_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyEval_EvalCodeWithName
> 	_PyFunction_FastCallDict
> 	_PyObject_Call_Prepend
> 	PyObject_Call
> 	_PyEval_EvalFrameDefault
> 	_PyEval_EvalCodeWithName
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	
> 	_PyMethodDef_RawFastCallKeywords
> 	_PyCFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyEval_EvalCodeWithName
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	
> 	_PyMethodDef_RawFastCallKeywords
> 	_PyCFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyEval_EvalCodeWithName
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	
> 	_PyMethodDef_RawFastCallKeywords
> 	_PyCFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyEval_EvalCodeWithName
> 	_PyFunction_FastCallDict
> 	_PyObject_Call_Prepend
> 	PyObject_Call
> 	_PyEval_EvalFrameDefault
> 	_PyGen_Send
> 	_PyMethodDef_RawFastCallKeywords
> 	_PyMethodDescr_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallKeywords
> 	_PyEval_EvalFrameDefault
> 	_PyFunction_FastCallDict
> *** End stack trace ***
> 

Here is the code for my LightningModule

```python
class GraphAEStage1(pl.LightningModule):
  def __init__(self, config):
    super(GraphAEStage1, self).__init__()
    max_deg = config['poc_max_degree']
    input_dim = config['input_dim']
    hidden_feat = config['hidden_feat_dim']
    layerid = config['layerid']
    self.lr = config['lr']
  
    self.encoder = GraphEncoderStage1(max_deg, input_dim, hidden_feat[layerid])
    self.decoder = GraphDecoderStage1(max_deg, input_dim, hidden_feat[layerid])
    
    self.encoder.apply(self.init_weights)
    
    self.crit = torch.nn.L1Loss()
    self.node_mae = MAE()
    self.adj_mae = MAE()
    
    self.decoder.self_linear.weight = nn.Parameter(self.encoder.self_linear.weight.permute(1,0))
    for d in range(config['poc_max_degree']):
        self.decoder.neigh_linear[d].weight = nn.Parameter(self.encoder.neigh_linear[d].weight.permute(1,0))

  def training_step(self, batch, batch_idx):
    loss, tf_node, _, _, _ = self.shared_step(batch)
    result = pl.TrainResult(minimize=loss)
    result.log('train_loss', loss, prog_bar = True, logger = True, on_step = False, on_epoch = True, sync_dist = True)
    return result

  def validation_step(self, batch, batch_idx):
    loss, tf_node, recon_node, val_node_mae, val_adj_mae = self.shared_step(batch)
    
    logs = {'val_loss': loss, 'node_mae': val_node_mae, 'adj_mae': val_adj_mae}
    result = pl.EvalResult(checkpoint_on = loss)
    result.log_dict(logs, prog_bar = True, logger = True, on_step = False, on_epoch = True, sync_dist = True)
    
    return result               

  def shared_step(self, batch):
    node, adj, deg, mask = batch
    tf_node, avg_adj = self.encoder(node, adj, deg, mask)
    recon_node, recon_adj = self.decoder(tf_node, deg)

    
    node_loss = self.criterion(recon_node, node)
    adj_loss = self.criterion(recon_adj, avg_adj)
    loss = node_loss + adj_loss
    
    node_mae = self.node_mae(node, recon_node)
    adj_mae = self.adj_mae(recon_adj, avg_adj)
    
    return loss, tf_node, recon_node, node_mae, adj_mae

  def configure_optimizers(self):
    optimizer = torch.optim.Adam(self.parameters(), lr=self.lr)
    return optimizer

  def criterion(self, output, target):
    return self.crit(output, target)

  def init_weights(self, m):
    if type(m) == nn.Linear:
        torch.nn.init.xavier_uniform(m.weight)
        m.bias.data.fill_(0.00)  
```

There are similar issues reported on PyTorch XLA repo, but many of them were using CrossEntropyLoss. I am not using CrossEntropyLoss here.

**Comment by ananyahjha93** [comment]
@Borda mind taking a look at this?

**Comment by lezwon** [comment]
@VirajBagal could you share the notebook?

**Comment by VirajBagal** [comment]
@lezwon Due to some privacy issues, I can't share it publicly here. I have shared it with you on Kaggle. Please check the "Shared With You" tab on Kaggle.

**Comment by lezwon** [comment]
Sure.. Will do 😊👍

**Comment by lezwon** [comment]
It seems xla does not support the `tensor.t()` operation on 1-dimensional tensors. @zcain117 could you help us out? `tensor([1,2], device=xm.xla_device()).t()` seems to be throwing the error mentioned when called.
